I am new to django-1.6. When I run the django server with DEBUG = True, it\'s running perfectly. But when I change DEBUG to False in the settings file, then the server stopped and it gives the following error on the command prompt:After I changed ALLOWED_HOSTS to ["http://127.0.0.1:8000",], in the browser I get the error:Is it possible to run Django without debug mode?The ALLOWED_HOSTS list should contain fully qualified host names, not urls. Leave out the port and the protocol. If you are using 127.0.0.1, I would add localhost to the list too:You could also use * to match any host:Quoting the documentation:Values in this list can be fully qualified names (e.g. \'www.example.com\'), in which case they will be matched against the request\xe2\x80\x99s Host header exactly (case-insensitive, not including port). A value beginning with a period can be used as a subdomain wildcard: \'.example.com\' will match example.com, www.example.com, and any other subdomain of example.com. A value of \'*\' will match anything; in this case you are responsible to provide your own validation of the Host header (perhaps in a middleware; if so this middleware must be listed first in MIDDLEWARE_CLASSES).Bold emphasis mine.The status 400 response you get is due to a SuspiciousOperation exception being raised when your host header doesn\'t match any values in that list.I had the same problem and i fixed it typing ALLOWED_HOSTS = [\'*\'] and to solve the problem with the static images you have to change the virtual paths in the enviroment configuration like this:Virtual Path                 Directory\n/static/                          /opt/python/current/app/yourpj/static/\n/media/                        /opt/python/current/app/Nuevo/media/I hope it helps you.PD: sorry for my bad english.For me, I got this error by not setting USE_X_FORWARDED_HOST to true.  From the docs: This should only be enabled if a proxy which sets this header is in use.My hosting service wrote explicitly in their documentation that this setting must be used, and I get this 400 error if I forget it.  With DEBUG = False in you settings file, you also need ALLOWED_HOST list set up.\nTry including ALLOWED_HOST = [\'127.0.0.1\', \'localhost\', \'www.yourdomain.com\']Otherwise you might receive a Bad Request(400) error from django.Navigate to settings and locate the base.py file\nSet the allowed hosts to \n ALLOWED_HOSTS = [\'*\']